### ISSUE TITLE
egfrd halts with gsl error

### ISSUE THREAD

**Issue by kaizu** [ecell/ecell4#127]
```
$ python3 profile.py vol egfrd 0
gsl: qag.c:248: ERROR: roundoff error prevents tolerance from being achieved
Default GSL error handler invoked.
Aborted
```

https://github.com/0ncorhynchus/ecell4-profile

**Comment by kaizu** [comment]
```python
from ecell4 import *

radius = 5e-9
L = 3.42e-6
N = 100000
M = int(min(L / (2 * radius), max(3, cbrt(N))))
rng = GSLRandomNumberGenerator()
rng.seed(2)

with species_attributes():
    A | { 'radius':  str(radius), 'D': '1.0e-12' }
model = get_model()

factory = egfrd.EGFRDFactory(Integer3(M, M, M), rng)
world = factory.create_world(Real3(L, L, L))
world.bind_to(model)
world.add_molecules(Species('A'), N)

simulator = factory.create_simulator(model, world)
for i in range(100):
    print(i, simulator.t())
    for j in range(N):
        simulator.step()
```

```
0 0.0
1 4.000000000000002e-09
2 1.7428284140456667e-06
3 3.3721804322537396e-06
4 5.216201909161823e-06
5 7.127036725496535e-06
6 8.649209388074504e-06
7 1.0447502884530218e-05
8 1.2168109183421203e-05
9 1.4001641199978884e-05
10 1.5713684696382152e-05
11 1.752443664650926e-05
12 1.9123913800180598e-05
13 2.094159530281295e-05
14 2.2779844236304562e-05
15 2.4730364293013058e-05
16 2.6457458450066752e-05
17 2.8264125277762576e-05
18 3.0214326056711892e-05
19 3.216300770707261e-05
20 3.404459327315172e-05
21 3.607745864783128e-05
gsl: qag.c:248: ERROR: roundoff error prevents tolerance from being achieved
Default GSL error handler invoked.
Aborted
```

**Comment by kaizu** [comment]
https://github.com/ecell/ecell4/blob/master/ecell4/egfrd/GreensFunction3DRadInf.cpp#L468-L473

```
gsl_integration_qag =>
tol = 4.45456e+13
n = 4
r = 1.11944e-08
t = 5.08006e-08
D = 2e-12
kf = 0
r0 = 1.0084e-08
Sigma = 1e-08
gsl: qag.c:248: ERROR: roundoff error prevents tolerance from being achieved
Default GSL error handler invoked.
Aborted
```

**Comment by kaizu** [comment]
By increasing the absolute tolerance for the integration (about three times larger), the error can be avoided.
https://github.com/ecell/ecell4/blob/master/ecell4/egfrd/GreensFunction3DRadInf.cpp#L648

However, I just modify the code to catch gsl error and throw it as a std error. You can catch the error even on Python.

```python
simulator = factory.create_simulator(model, world)
for i in range(N):
    try:
        simulator.step()
    except RuntimeError as err:
        print(err)
        break
```

**Comment by kaizu** [comment]
I got the reason. No error was raised there with epdp. The following causes this error:
https://github.com/ecell/ecell4/blob/master/ecell4/egfrd/compat.h#L28-L34
After commenting out the definition, the error was avoided.
However, on the environment with no `sincos` function, the same error would be raised.

**Comment by kaizu** [comment]
After replacing `SphericalBesselTable.hpp` with the one generated by `epdp`, no error was raised. This problem seems due to the accuracy in `SphercialBesselTable.hpp`.

**Comment by kaizu** [comment]
The following figure shows the relative difference in `sj_table4_f` between `epdp` and `ecell4`.
![output](https://cloud.githubusercontent.com/assets/778255/15528673/fd54738c-2280-11e6-94c6-df7d1fdb74ed.png)

**Comment by kaizu** [comment]
Fixed by 1b623de.
